src/jit/lsra.cpp

```diff
@@ -5171,22 +5206,71 @@ void LinearScan::unassignPhysReg( RegRecord * regRec, RefPosition* spillRefPosit
 
     if (spillRefPosition == nullptr)
     {
-        assert(!assignedInterval->isActive || assignedInterval->physReg != thisRegNum);
+        // Note that we can't assert  for the copyReg case
+        //
+        if (assignedInterval->physReg == thisRegNum)
+        {
+            assert(assignedInterval->isActive == false);
+        }
     }
     else
     {
         assert(spillRefPosition->getInterval() == assignedInterval);
     }
+
     regRec->assignedInterval = nullptr;
+}
+
+//------------------------------------------------------------------------
+// unassignPhysReg: Unassign the given physical register record, and spill the
+//                  assignedInterval at the given spillRefPosition, if any.
+//
+// Arguments:
+//    regRec           - the RegRecord to be unasssigned
+//    spillRefPosition - The RefPosition at which the assignedInterval is to be spilled
+//
+// Return Value:
+//    None.
+//
+// Assumptions:
+//    The assignedInterval must not be null.
+//    If spillRefPosition is null, the assignedInterval must be inactive, or not currently
+//    assigned to this register (e.g. this is a copyReg for that Interval).
+//    Otherwise, spillRefPosition must be associated with the assignedInterval.
+//
+void LinearScan::unassignPhysReg( RegRecord * regRec, RefPosition* spillRefPosition)
+{
+    Interval * assignedInterval = regRec->assignedInterval;
+    assert(assignedInterval != nullptr);
+    checkAndClearInterval(regRec, spillRefPosition);
+    regNumber thisRegNum = regRec->regNum;
 
 #ifdef _TARGET_ARM_
     if ((assignedInterval->registerType == TYP_DOUBLE) && isFloatRegType(regRec->registerType))
     {
-        RegRecord * nextRegRec = getRegisterRecord(REG_NEXT(regRec->regNum));
-        nextRegRec->assignedInterval = nullptr;
+        regNumber   nextRegNum = REG_NEXT(regRec->regNum);
+        RegRecord * nextRegRec = getRegisterRecord(nextRegNum);
+        checkAndClearInterval(nextRegRec, spillRefPosition);
     }
 #endif // _TARGET_ARM_
 
+#if FEATURE_MULTIREG_ARGS_OR_RET
+    if (assignedInterval->registerType == TYP_STRUCT)
```

Comment:
I moved all of the code that checks for and handles the TYP_STRUCT case to be under #if TARGET_ARM64